src/me/corriekay/pokegoutil/utils/pokemon/PokemonCalculationUtils.java

```diff
@@ -308,41 +324,43 @@ public static double weaveDps(final PokemonId pokemonId, final PokemonMove move1
         //=IF(AB2=100,CEILING(AB2/U2),AB2/U2)
         final double weaveEnergyUsageRatio;
 
-        if (Math.abs(pm2.getEnergy()) == MAX_MOVE_ENERGY) {
-            weaveEnergyUsageRatio = Math.ceil((double) Math.abs(pm2.getEnergy()) / (double) pm1.getEnergy());
-        } else {
-            weaveEnergyUsageRatio = (double) Math.abs(pm2.getEnergy()) / (double) pm1.getEnergy();
-        }
+        if (pm1 != null && pm2 != null) {
```

Comment:
It's a style preference.  My employer has strict rules on only one return from a method for code maintenance, so it has snuck its way into my coding.